Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Data/ExportImport2/ProductExportPagedDataSource.cs

```diff
@@ -62,18 +62,16 @@ protected override ExportableSearchResult FetchData(ProductSearchCriteria search
         protected virtual IEnumerable<IExportable> ToExportable(IEnumerable<ICloneable> objects)
         {
             var models = objects.Cast<CatalogProduct>();
-            var viewableMap = models.ToDictionary(x => x, x => AbstractTypeFactory<ExportableProduct>.TryCreateInstance().FromModel(x));
+            var modelIds = models.Select(x => x.Id).ToList();
 
-            foreach (var kvp in viewableMap)
+            var exportableProducts = models.Select(x =>
             {
-                var model = kvp.Key;
-                var viewableEntity = kvp.Value;
-                viewableEntity.ImageUrl = model.ImgSrc;
-            }
-
-            var modelIds = models.Select(x => x.Id).ToList();
+                var exportableProduct = AbstractTypeFactory<ExportableProduct>.TryCreateInstance().FromModel(x);
+                exportableProduct.ImageUrl = x.ImgSrc;
+                return exportableProduct;
+            }).OrderBy(x => modelIds.IndexOf(x.Id));
```

Comment:
We do not need ordering here, it was done to restore order after creating dictionary. And no need in modelds variable